pymtl3/stdlib/connects/connect_bits2bitstruct.py

```diff
@@ -0,0 +1,63 @@
+'''
+==========================================================================
+connect_bits2bitstruct.py
+==========================================================================
+A connect function that connects a bits signal and a bitsrtuct signal that
+has the same width.
+
+Author : Yanghui Ou
+  Date : Feb 24, 2020
+'''
+from pymtl3 import Bits, connect
+from pymtl3.datatypes.bitstructs import _FIELDS, is_bitstruct_class
+
+#-------------------------------------------------------------------------
+# _connect_bits2bitstruct_h
+#-------------------------------------------------------------------------
+# Helper function for connect_bits2bitstruct.
+
+def _connect_bits2bitstruct_h( field, bits_signal, slices ):
+
+  # NOTE: We do not support list of fields at the moment because: 1) there
+  # are very few use cases; 2) the pack order of a list of fields is
+  # somewhat ambiguous.
+  if isinstance( field, list ):
+    assert False, "A list of fields is not supported at the moment."
+
+  field_type = field.get_type()
+
+  if issubclass( field_type, Bits ):
+    start     = 0 if not slices else slices[-1].stop
+    stop      = start + field_type.nbits
+    cur_slice = slice( start, stop )
```

Comment:
do you actually need to carry a list of slices? I guess you only need to record the last slice stop?

Reply:
Ideally I only need one integer, but after a recursive call changes the integer value it needs to be observable by the parent call. Originally I use a single element list to do this but it turns out to be a little confusing. If it is C++ I would just record an integer pointer...